velox/parse/TypeResolver.cpp

```diff
@@ -52,6 +52,73 @@ std::string toString(
   return out.str();
 }
 
+std::pair<uint8_t, uint8_t> computeRoundDecimalResultPrecisionScale(
+    const uint8_t aPrecision,
+    const uint8_t aScale,
+    const int32_t scale) {
+  const auto& integralLeastNumDigits = aPrecision - aScale + 1;
+  if (scale < 0) {
+    const auto& newPrecision = std::max(integralLeastNumDigits, -aScale + 1);
+    return {std::min(newPrecision, 38), 0};
+  } else {
+    return {
+        std::min(
+            static_cast<int32_t>(
+                integralLeastNumDigits +
+                std::min(static_cast<int32_t>(aScale), scale)),
+            38),
+        std::min(static_cast<int32_t>(aScale), scale)};
+  }
+}
+
+// int32_t evaluate(core::TypedExprPtr input) {
+//   std::shared_ptr<memory::MemoryPool> pool =
+//   memory::addDefaultLeafMemoryPool(); std::shared_ptr<core::QueryCtx>
+//   queryCtx_{std::make_shared<core::QueryCtx>()};
+//   std::unique_ptr<core::ExecCtx> execCtx_{
+//       std::make_unique<core::ExecCtx>(pool.get(), queryCtx_.get())};
+//   exec::ExprSet exprSet({input}, execCtx_.get());
+//   auto vector = BaseVector::createNullConstant(INTEGER(), 1, pool.get());
+//   // RowVectorPtr data = RowVector::createEmpty(ROW({INTEGER()}),
+//   pool.get()); RowVectorPtr data = std::make_shared<RowVector>(
+//       pool.get(), ROW({INTEGER()}), BufferPtr(nullptr), 1, {vector});
+//   exec::EvalCtx evalCtx(execCtx_.get(), &exprSet, vector);
+//   std::vector<VectorPtr> results(1);
+//   exprSet.eval(SelectivityVector{0}, evalCtx, results);
+//   return results[0]->asUnchecked<ConstantVector<int32_t>>()->valueAt(0);
+// }
+
+// int32_t getRoundDecimalScale(core::TypedExprPtr input) {
+//   return evaluate(input);
+// }
+
+int32_t getRoundDecimalScale(core::TypedExprPtr input) {
```

Comment:
Here we get the second argument `scale`, now it just match the test expression `cast 30 as integer`. How can I eval with expression to get the constant value? Can you give me some clue? Thanks!  @mbasmanova 
The above commented function needs a data vector to get the result.

Reply:
@jinchengchenghh I'm not sure I fully understand the question. It might be easier if you could add a simple test case to DuckParserTest and explain what you expect to get and how it differs from what you are getting.

You may want to look into setting ParseOptions::parseIntegerAsBigint to false and remove 'cast' from the expression: "decimal_round(c0, 30)".

Reply:
Now the test can pass with current code, but if the test changed to `cast string as integer`, it won't work well. So I need to know how to pass all the expression that will compute the second argument scale.

Reply:
I do not understand why you want to use 'cast string as integer' and not an integer constant in the expression. Would you clarify?

Reply:
Does the user may write as this?

Reply:
Can we only support `decimal_round(c0, cast(30 as integer)` or `decimal_round(c0, 30)`?

Reply:
Velox doesn't support SQL inputs. We use SQL expressions to write test only for convenience.